src/main/java/tsunami/security/scanner/utilities/FreeMarkerUtil.java

```diff
@@ -40,20 +40,45 @@ private FreeMarkerUtil() throws IOException {
     cfg.setFallbackOnNullLoopVariable(false);
   }
 
+  public static String replaceTemplates(Map<String, String> templateDataMap, File configFile)
+      throws IOException, TemplateException {
+
+    // Split input configFile into path and file name.
+    String file = configFile.getName();
+    String filePath = configFile.getPath();
+    String path = filePath.substring(0, filePath.lastIndexOf("/"));
+    System.out.println("file: " + file);
+    System.out.println("path: " + path);
+
+    // Set path for config templates
+    cfg.setDirectoryForTemplateLoading(new File(path));
+
+    // Get the template file.
+    Template temp = cfg.getTemplate(file);
+
+    StringWriter stringWriter = new StringWriter();
+    temp.process(templateDataMap, stringWriter);
+
+    // get the config in String format from the StringWriter
+    String config = stringWriter.toString();
+    System.out.println(config);
+
+    return config;
+  }
+
+  // Deprecated
   public static String replaceTemplates(String version, String password, File configFile)
```

Comment:
Is this method used anywhere else? If not, you can simply remove it. If you plan to keep it, then maybe call the other `replaceTemplates(Map<String, String> templateDataMap, File configFile)` method after you build the `templateDataMap`.